get-api/get-api.go

```diff
@@ -0,0 +1,72 @@
+package getapi
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	httpclient "thien.com/http-client"
+)
+
+const (
+	getPostsEndpoint    = "https://my-json-server.typicode.com/typicode/demo/posts"
+	getCommentsEndpoint = "https://my-json-server.typicode.com/typicode/demo/comments"
+)
+
+type Post struct {
+	ID    int64  `json:"id"`
+	Title string `json:"title"`
+}
+type Comment struct {
+	ID     int64  `json:"id"`
+	Body   string `json:"body"`
+	PostID int64  `json:"postId"`
+}
+
+type GetApi interface {
+	GetPosts() ([]Post, error)
+	GetComments() ([]Comment, error)
+}
+
+type getApiImpl struct {
+	httpClient httpclient.HTTPClient
+}
+
+func (api *getApiImpl) GetPosts() ([]Post, error) {
```

Comment:
Nitpick: this function does 2 things, get data and rendering, we can separate it into 2 smaller functions